src/serializer.ts

```diff
@@ -17,14 +19,157 @@ interface PaymentJSON {
  * Provides functionality to serialize from protocol buffers to JSON objects.
  */
 class Serializer {
+/**
+   * Convert a Transaction to a JSON representation.
+   *
+   * @param {proto.Transaction} transaction A Transaction to convert.
+   * @returns {Object} The Transaction as JSON.
+   */
+  public static transactionToJSON(
+    transaction: Transaction
+  ): object | undefined {
+    // Serialize the protocol buffer to a JSON representation.
+    var object: any = transaction.toObject();
+
+    // Convert fields names where direct conversion is possible.
+    this.convertPropertyName("sequence", "Sequence", object);
+    this.convertPropertyName("signingPublicKey", "SigningPubKey", object);
+    this.convertPropertyName(
+      "lastLedgerSequence",
+      "LastLedgerSequence",
+      object
+    );
+
+    // Delete unsupported fields from the protocol buffer.
+    delete object.memosList;
+    delete object.flags;
+    delete object.signature;
+    delete object.signersList;
+    delete object.sourceTag;
+    delete object.accountTransactionId;
+
+    // Encode SigningPubKey to hex, which is what ripple-binary-codec expects.
+    object.SigningPubKey = Utils.toHex(transaction.getSigningPublicKey_asU8());
+
+    // Convert account field, handling X-Addresses if needed.
+    const accountAddress = transaction.getAccount();
+    if (!accountAddress) {
+      return;
+    }
+    const account = accountAddress.getAddress();
```

Comment:
If `account` is an address, what is `accountAddress`?

Reply:
`AccountAddress` is a special protocol buffer that wraps addresses. This gives us slightly better typing. Imagine:

```
message AccountInfo {
   string address = 1;
}
```

vs

```
message AccountAddress {
  string address = 1;
}

message AccountInfo {
  AccountAddress address = 1;
}
```

In the former case, you can stuff anything you want in the `string` field, where in the later, we've used a wrapper type to make it clear what specific KIND of string should be placed in it.